apps/server/src/modules/oauth/uc/oauth.uc.spec.ts

```diff
@@ -65,37 +61,83 @@ describe('OAuthUc', () => {
 
-		service = await module.resolve<OauthUc>(OauthUc);
-		oauthService = await module.resolve<OAuthService>(OAuthService);
+		oauthService = await module.get(OAuthService);
+		uc = await module.get(OauthUc);
+		defaultSystem = systemFactory.withOauthConfig().build();
+		defaultOauthConfig = defaultSystem.oauthConfig as OauthConfig;
+
+		defaultAuthCode = '43534543jnj543342jn2';
+		defaultQuery = { code: defaultAuthCode };
+		defaultErrorQuery = { error: 'oauth_login_failed' };
+		defaultScool = schoolFactory.build();
+		defaultDecodedJWT = {
+			sub: '4444',
+			uuid: '1111',
+		};
+
+		defaultUser = {
+			email: '',
+			roles: new Collection<Role>([]),
+			school: defaultScool,
+			_id: new ObjectId(),
+			id: '4444',
+			createdAt: new Date(),
+			updatedAt: new Date(),
+			ldapId: '1111',
+			firstName: 'Test',
+			lastName: 'Testmann',
+		};
+		defaultJWT =
+			'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJ1dWlkIjoiMTIzIn0.H_iI0kYNrlAUtHfP2Db0EmDs4cH2SV9W-p7EU4K24bI';
+		defaultTokenResponse = {
+			access_token: 'zzzz',
+			refresh_token: 'zzzz',
+			id_token: defaultJWT,
+		};
+
+		defaultResponse = {
+			idToken: defaultJWT,
+			logoutEndpoint: 'logoutEndpointMock',
+			provider: 'providerMock',
+			redirect: '',
+		};
+		defaultErrorRedirect = `${Configuration.get('HOST') as string}/login?error=${
+			defaultErrorQuery.error as string
+		}&provider=${defaultOauthConfig.provider}`;
+		defaultErrorResponse = new OAuthResponse();
+		defaultErrorResponse.errorcode = defaultErrorQuery.error as string;
+		defaultErrorResponse.redirect = defaultErrorRedirect;
+
+		oauthService.checkAuthorizationCode.mockReturnValue(defaultAuthCode);
+		oauthService.getOauthConfig.mockResolvedValue(defaultOauthConfig);
+		oauthService.requestToken.mockResolvedValue(defaultTokenResponse);
+		oauthService.validateToken.mockResolvedValue(defaultDecodedJWT);
+		oauthService.findUser.mockResolvedValue(defaultUser);
+		oauthService.getJwtForUser.mockResolvedValue(defaultJWT);
+		oauthService.buildResponse.mockReturnValue(defaultResponse);
+		oauthService.getRedirect.mockReturnValue(defaultResponse);
 	});
 
 	it('should be defined', () => {
-		expect(service).toBeDefined();
+		expect(uc).toBeDefined();
 	});
 
-	describe('startOauth', () => {
-		it('should start the OAuth2.0 process and get a redirect ', async () => {
-			jest.spyOn(oauthService, 'processOAuth').mockResolvedValue(defaultIservResponse);
-			const response = await service.startOauth(defaultQuery, defaultIservSystemId);
-			expect(response).toEqual(defaultIservResponse);
-		});
-		it('should start the OAuth2.0 process and get a redirect ', async () => {
-			jest.spyOn(oauthService, 'processOAuth').mockResolvedValue(defaultResponse);
-			const response = await service.startOauth(defaultQuery, defaultSystemId);
-			expect(response).toEqual(defaultIservResponse);
+	describe('processOAuth', () => {
+		it('should do the process', async () => {
```

Comment:
this seems like very little testing for such a central and involved orchestrating function...

Reply:
processOAuth "just" orchestrates all the other functions in the oauth flow. these functions are all tested seperately. I should suffice to give this functions starting parameters and expect the expected outcome. the rest ist tested elsewhere.